lambdas/call-ticf-cri/src/main/java/uk/gov/di/ipv/core/callticfcri/CallTicfCriHandler.java

```diff
@@ -177,6 +136,74 @@ public Map<String, Object> handleRequest(ProcessRequest input, Context context)
+        } finally {
+            if (ipvSessionItem != null) {
+                ipvSessionService.updateIpvSession(ipvSessionItem);
+            }
         }
     }
+
+    @Tracing
+    private Map<String, Object> callTicfCri(IpvSessionItem ipvSessionItem, ProcessRequest request)
+            throws ParseException, TicfCriServiceException, CiRetrievalException, SqsException,
+                    VerifiableCredentialException, CiPostMitigationsException, CiPutException,
+                    JsonProcessingException {
+        String featureSet = RequestHelper.getFeatureSet(request);
+        configService.setFeatureSet(featureSet);
+        ClientOAuthSessionItem clientOAuthSessionItem =
+                clientOAuthSessionDetailsService.getClientOAuthSession(
+                        ipvSessionItem.getClientOAuthSessionId());
+        LogHelper.attachGovukSigninJourneyIdToLogs(
+                clientOAuthSessionItem.getGovukSigninJourneyId());
+
+        List<String> vcToSendToTicf = getVcToSendToTicf(clientOAuthSessionItem, ipvSessionItem);
+
+        List<SignedJWT> ticfVcs =
+                ticfCriService.getTicfVc(clientOAuthSessionItem, ipvSessionItem, vcToSendToTicf);
+
+        if (ticfVcs.isEmpty()) {
+            LogHelper.logMessage(Level.INFO, "No VC to process - returning next");
+            return JOURNEY_NEXT;
+        }
+
+        criStoringService.storeVcs(
+                TICF_CRI, request.getIpAddress(), ticfVcs, clientOAuthSessionItem, ipvSessionItem);
+
+        ContraIndicators cis =
+                ciMitService.getContraIndicatorsVC(
+                        clientOAuthSessionItem.getUserId(),
+                        clientOAuthSessionItem.getGovukSigninJourneyId(),
+                        request.getIpAddress());
+
+        if (ciMitUtilityService.isBreachingCiThreshold(cis)) {
+            LogHelper.logMessage(Level.INFO, "CI score is breaching threshold - setting VOT to P0");
+            ipvSessionItem.setVot(VOT_P0);
+
+            return JOURNEY_FAIL_WITH_CI;
+        }
+
+        LogHelper.logMessage(Level.INFO, "CI score not breaching threshold");
+        return JOURNEY_NEXT;
+    }
+
+    @Tracing
+    private List<String> getVcToSendToTicf(
+            ClientOAuthSessionItem clientOAuthSessionItem, IpvSessionItem ipvSessionItem) {
+        List<String> vcInStore =
+                userIdentityService.getUserIssuedCredentials(
+                        verifiableCredentialService.getVcStoreItems(
+                                clientOAuthSessionItem.getUserId()));
+        List<String> vcToSendToTicf = new ArrayList<>();
+        List<String> vcReceivedThisSession = ipvSessionItem.getVcReceivedThisSession();
+
+        if (vcReceivedThisSession != null && !vcReceivedThisSession.isEmpty()) {
+            for (String storeVc : vcInStore) {
+                if (vcReceivedThisSession.contains(storeVc)) {
+                    vcToSendToTicf.add(storeVc);
+                }
+            }
+        }
```

Comment:
Is there a reason not to use streams here? I think it makes the intent a bit clearer:
```
return vcInStore.stream().filter(vcReceivedThisSession::contains).toList(); // or the expanded lambda syntax
```
(You'll still need the null check, but don't really need the empty check)

Reply:
No, no reason. Will switch it up.